pdal/private/MathUtils.cpp

```diff
@@ -384,5 +384,55 @@ void writeMatrix(Eigen::MatrixXd data, const std::string& filename,
 }
 #pragma warning (pop)
 
+
+// https://en.wikipedia.org/wiki/Barycentric_coordinate_system
+// http://blackpawn.com/texts/pointinpoly/default.html
+//
+/// Return the interpolated Z value at location X/Y. If the X/Y location is not
+/// in the triangle, return infinity.  If the determinant is 0, the input points
+/// aren't a triangle (they're collinear).
+/// \param x1, y1, z1  Coordinates of point 1.
+/// \param x2, y2, z2  Coordinates of point 2.
+/// \param x3, y3, z3  Coordinates of point 3.
+/// \param x, y  X and Y coordinates of location to find an interpolated Z
+/// \return  Interpolated Z value or infinity.
+double barycentricInterpolation(double x1, double y1, double z1,
```

Comment:
I'm guessing you didn't change anything when moving to MathUtils (good call doing so BTW). We don't have any tests for this though. Would probably be a good thing to add going forward.

Reply:
There are tests in hag that exercise this.   I could extract, I think.

Reply:
Right, I suppose by proxy via https://github.com/PDAL/PDAL/blob/master/test/unit/filters/HAGFilterTest.cpp#L47.